plugins/outputs/yandex_cloud_monitoring/yandex_cloud_monitoring.go

```diff
@@ -0,0 +1,276 @@
+	IAMTokenFromConfig string            `toml:"iam_token"`
+
+	IAMToken               string
+	IamTokenExpirationTime time.Time
+
+	client *http.Client
+
+	timeFunc func() time.Time
+
+	MetricOutsideWindow selfstat.Stat
+}
+
+type yandexCloudMonitoringMessage struct {
+	TS      string                        `json:"ts,omitempty"`
+	Labels  map[string]string             `json:"labels,omitempty"`
+	Metrics []yandexCloudMonitoringMetric `json:"metrics"`
+}
+
+type yandexCloudMonitoringMetric struct {
+	Name       string            `json:"name"`
+	Labels     map[string]string `json:"labels"`
+	MetricType string            `json:"type,omitempty"` // DGAUGE|IGAUGE|COUNTER|RATE. Default: DGAUGE
+	TS         string            `json:"ts,omitempty"`
+	Value      float64           `json:"value"`
+}
+
+type MetadataIamToken struct {
+	AccessToken string `json:"access_token"`
+	ExpiresIn   int64  `json:"expires_in"`
+	TokenType   string `json:"token_type"`
+}
+
+const (
+	defaultRequestTimeout    = time.Second * 5
+	defaultEndpointUrl       = "https://monitoring.api.cloud.yandex.net/monitoring/v2/data/write"
+	defaultMetadataTokenUrl  = "http://169.254.169.254/computeMetadata/v1/instance/service-accounts/default/token"
+	defaultMetadataFolderUrl = "http://169.254.169.254/computeMetadata/v1/instance/attributes/folder-id"
+	maxRequestBodySize       = 4000000
+)
+
+var sampleConfig = `
+  ## Timeout for HTTP writes.
+  # timeout = "20s"
+
+  ## Normally should not be changed
+  # endpoint_url = "https://monitoring.api.cloud.yandex.net/monitoring/v2/data/write"
+
+  ## Normally folder ID is taken from Compute instance metadata
+  # folder_id = "..."
+
+  ## Can be set explicitly for authentification debugging purposes 
+  # iam_token = "..."  
+`
+
+// Description provides a description of the plugin
+func (a *YandexCloudMonitoring) Description() string {
+	return "Send aggregated metrics to Yandex.Cloud Monitoring"
+}
+
+// SampleConfig provides a sample configuration for the plugin
+func (a *YandexCloudMonitoring) SampleConfig() string {
+	return sampleConfig
+}
+
+// Connect initializes the plugin and validates connectivity
+func (a *YandexCloudMonitoring) Connect() error {
+	if a.Timeout.Duration <= 0 {
+		a.Timeout.Duration = defaultRequestTimeout
+	}
+
+	a.client = &http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+		},
+		Timeout: a.Timeout.Duration,
+	}
+
+	if a.EndpointUrl == "" {
+		a.EndpointUrl = defaultEndpointUrl
+	}
+	if a.MetadataTokenUrl == "" {
+		a.MetadataTokenUrl = defaultMetadataTokenUrl
+	}
+	if a.MetadataFolderUrl == "" {
+		a.MetadataFolderUrl = defaultMetadataFolderUrl
+	}
+	if a.FolderID == "" {
+		folderID, err := getFolderIDFromMetadata(a.client, a.MetadataFolderUrl)
+		if err != nil {
+			return err
+		}
+		a.FolderID = folderID
+	}
+	if a.Service == "" {
+		a.Service = "custom"
+	}
+
+	log.Printf("D! Writing to Yandex.Cloud Monitoring URL: %s", a.EndpointUrl)
```

Comment:
If you add `Log telegraf.Logger `toml:"-"` to the struct, you can use the Telegraf logger that knows about plugin names and log levels. In tests, set it to `telegraf.Logger{}`

Reply:
Done